src/plugins/intel_gpu/src/graph/eltwise.cpp

```diff
@@ -107,6 +107,130 @@ layout eltwise_inst::calc_output_layout(eltwise_node const& node, kernel_impl_pa
+                                                       eltwise_mode::mod,
+                                                       eltwise_mode::eq,
+                                                       eltwise_mode::ne,
+                                                       eltwise_mode::lt,
+                                                       eltwise_mode::le,
+                                                       eltwise_mode::gt,
+                                                       eltwise_mode::ge,
+                                                       eltwise_mode::squared_diff,
+                                                       eltwise_mode::floor_mod,
+                                                       eltwise_mode::logic_and,
+                                                       eltwise_mode::logic_or,
+                                                       eltwise_mode::logic_xor};
+        if (std::find(eltwise_int_modes.begin(), eltwise_int_modes.end(), mode) == eltwise_int_modes.end())
+            CLDNN_ERROR_MESSAGE(desc->id, "Requested eltwise mode is not supported for integer types.");
+    }
+
+    // Logic and comparison operations should return i8 for any inputs
+    std::vector<eltwise_mode> eltwise_bool_modes = {eltwise_mode::eq,
+                                                    eltwise_mode::ne,
+                                                    eltwise_mode::lt,
+                                                    eltwise_mode::le,
+                                                    eltwise_mode::gt,
+                                                    eltwise_mode::ge,
+                                                    eltwise_mode::logic_and,
+                                                    eltwise_mode::logic_or,
+                                                    eltwise_mode::logic_xor};
+    if (std::find(eltwise_bool_modes.begin(), eltwise_bool_modes.end(), mode) != eltwise_bool_modes.end()) {
+        output_layout.data_type = data_types::i8;
+    }
+
+    if (desc->output_data_type) {
+        output_layout.data_type = *desc->output_data_type;
+    }
+
+    if (impl_param.has_fused_primitives()) {
+        output_layout.data_type = impl_param.get_fused_output_layout().data_type;
+    }
+
+    if (!desc->stride.empty()) {
+        auto new_size = input_node_layout.get_tensor();
+        // we can safely use only first stride, since we're using first input, and input / stride should give exact same
+        // value for every input
+        new_size.spatial[0] = (input_node_layout.spatial(0) - 1) / desc->stride[0].spatial[0] + 1;
+        new_size.spatial[1] = (input_node_layout.spatial(1) - 1) / desc->stride[0].spatial[1] + 1;
+        new_size.spatial[2] = (input_node_layout.spatial(2) - 1) / desc->stride[0].spatial[2] + 1;
+        input_node_layout.set_tensor(new_size);
+        return { input_node_layout };
+    }
+    return { output_layout };
+}
+
```

Comment:
This is stilll legacy impl. Please call ngraph::shape_infer except when legacy case is needed.

Reply:
In general we can have shape infer impl which doesn't call ngraph for trivial cases.  This op may also have own impl as ngraph shape infer call will require split for different eltwise modes.

Reply:
I see. But I think we should let them to be sync-d with the original impl. @ahnyoung-paul  Please add some comments where the original impl is from.

Reply:
Add comments where the original impl is from